### PR TITLE
Handle synthetic .apply in Database.names, fixes #837

### LINKED ISSUE #837: Unexpected symbol for desugared .apply
The code
```scala
object tup {
  val foo = (a: (Int, Boolean)) => a
  foo(2, true)
}
```
has the following semanticdb
```
Names:
[8..30): autotuplingwithlambdas => _root_.autotuplingwithlambdas.
[40..43): tup => _root_.autotuplingwithlambdas.tup.
[52..55): foo => _root_.autotuplingwithlambdas.tup.foo.
[59..60): a => _root_.autotuplingwithlambdas.tup.foo.$anonfun.(a)
[63..66): Int => _root_.scala.Int#
[68..75): Boolean => _root_.scala.Boolean#
[81..82): a => _root_.autotuplingwithlambdas.tup.foo.$anonfun.(a)
[85..88): foo => _root_.scala.Function1#apply(Ljava/lang/Object;)Ljava/lang/Object;. // expected Symbol.Multi with _root_.autotuplingwithlambdas.tup.foo.

Denotations:
_root_.autotuplingwithlambdas. => package autotuplingwithlambdas
_root_.autotuplingwithlambdas.tup. => final object tup
_root_.autotuplingwithlambdas.tup.foo. => val foo: ((Int, Boolean)) => (Int, Boolean)
_root_.autotuplingwithlambdas.tup.foo.$anonfun.(a) => param a: (Int, Boolean)

Sugars:
[97..97) [Int, Boolean]
```

It would be great if we could query from the call-site `foo` and get the denotation `_root_.autotuplingwithlambdas.tup.foo. => val foo: ((Int, Boolean)) => (Int, Boolean)`

Discovered in https://github.com/scalacenter/scalafix/pull/143

### PR DESCRIPTION
**Pull request by olafurpg**
The synthetic apply belongs to Database.sugars, see
https://github.com/scalameta/scalameta/issues/846

Review @gabro

### CODE REVIEW

**Comment by gabro** [comment]
If I get this right, given

```scala
val foo = (x: Int) => x
foo(42)
```

when we look up the symbol `foo` at call site, now we get `foo` instead of `apply`?

File: scalahost/nsc/src/test/scala/scala/meta/tests/scalahost/mirrors/SemanticSuite.scala [hunk 1]
```diff
@@ -235,4 +235,24 @@ class SemanticSuite extends DatabaseSuite {
...
+       |  foo.apply(2, true)
+       |}
+    """.stripMargin,
+    """
+      |[8..11): tup => _empty_.tup.
+      |[20..23): foo => _empty_.tup.foo.
+      |[27..28): a => _empty_.tup.foo.$anonfun.(a)
+      |[31..34): Int => _root_.scala.Int#
+      |[36..43): Boolean => _root_.scala.Boolean#
+      |[53..56): foo => _empty_.tup.foo.
```
Comment (olafurpg) [inline scalahost/nsc/src/test/scala/scala/meta/tests/scalahost/mirrors/SemanticSuite.scala]: Correct @gabro, we use the Database.syntax to test this here.

Comment (gabro) [inline scalahost/nsc/src/test/scala/scala/meta/tests/scalahost/mirrors/SemanticSuite.scala]: 👍 cool!

File: scalahost/nsc/src/main/scala/scala/meta/internal/semantic/AttributesOps.scala [hunk 2]
```diff
@@ -333,6 +335,11 @@ trait AttributesOps { self: DatabaseOps =>
     }
   }
 
+  private lazy val applyCharacters = "apply".toCharArray
+  private def isSyntheticApply(select: g.Select): Boolean =
+    select.pos == select.qualifier.pos &&
+      select.name.toChars.sameElements(applyCharacters)
```
Comment (xeno-by) [inline scalahost/nsc/src/main/scala/scala/meta/internal/semantic/AttributesOps.scala]: `select.name == g.nme.apply`

Comment (olafurpg) [inline scalahost/nsc/src/main/scala/scala/meta/internal/semantic/AttributesOps.scala]: Cool! There had to be something better ^^

**Pull request merged**